src/schema-generator/types.ts

```diff
@@ -220,7 +222,7 @@ export type TComponentSchema<V = undefined> =
 	| IWrapperSchema
 	| IElementSchema
 	| ICustomComponentSchema;
-export type TFieldValidation = TFieldSchema["validation"];
+export type TFieldValidation = Exclude<TFieldSchema, IErrorFieldSchema>["validation"];
```

Comment:
I think this is not needed, instead the rest of the custom rules in schemas should extend from `IYupValidation` (looks like we really missed this out)